aws_lambda_builders/workflows/nodejs_npm/actions.py

```diff
@@ -111,3 +111,87 @@ def execute(self):
 
         except NpmExecutionError as ex:
             raise ActionFailedError(str(ex))
+
+class NodejsNpmrcCopyAction(BaseAction):
+
+    """
+    A Lambda Builder Action that copies NPM config file .npmrc
+    """
+
+    NAME = 'CopyNpmrc'
+    DESCRIPTION = "Copying configuration from .npmrc"
+    PURPOSE = Purpose.COPY_SOURCE
+
+    def __init__(self, artifacts_dir, source_dir, osutils):
+        """
+        :type artifacts_dir: str
+        :param artifacts_dir: an existing (writable) directory with project source files.
+            Dependencies will be installed in this directory.
+
+        :type source_dir: str
+        :param source_dir: directory containing project source files.
+
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+        """
+
+        super(NodejsNpmrcCopyAction, self).__init__()
+        self.artifacts_dir = artifacts_dir
+        self.source_dir = source_dir
+        self.osutils = osutils
+
+    def execute(self):
+        """
+        Runs the action.
+
+        :raises lambda_builders.actions.ActionFailedError: when .npmrc copying fails
+        """
+
+        try:
+            npmrc_path = self.osutils.joinpath(self.source_dir, ".npmrc")
+            if self.osutils.file_exists(npmrc_path):
+                LOG.debug(".npmrc copying in: %s", self.artifacts_dir)
+                self.osutils.copy_file(npmrc_path, self.artifacts_dir)
+
+        except OSError as ex:
+            raise ActionFailedError(str(ex))
+
+class NodejsCleanUpAction(BaseAction):
```

Comment:
This name indicates we are cleaning up more than `.npmrc`. Can we scope the name down to be Npmrc, like how you have done for the Copy Action? Or was the goal to make one Npm Clean up action? 

Side thought: I wonder if this will be a pattern that develops (adding then removing the same set of files), to the point where it's worth codifying.

Reply:
there will be some other clean up actions to add later (such as `npm dedupe`). Would you prefer each step to be a separate action, or do we keep it all inside a single action?

on a related note, is it worth adding another `Purpose` value there? This isn't really `COPY_SOURCE`, but nothing else fit.